mne/io/nedf/nedf.py

```diff
@@ -0,0 +1,181 @@
+    info['sfreq'] = int(eegset.find('EEGSamplingRate').text)
+    info['ch_names'] = [e.text for e in eegset.find('EEGMontage')]
+    # expect nchantotal uint24s
+    datadt.append(('eeg', 'B', (nchantotal, 3)))
+
+    info['units'] = eegset.find('EEGUnits')
+
+    if headerxml.find('STIMSettings'):
+        # 2* -> two stim samples per eeg sample
+        datadt.append(('stim', 'B', (2, nchantotal, 3)))
+
+    if 'AdditionalChannelStatus' in headerxml:
+        raise RuntimeError('Unexpected AdditionalChannelStatus')
+
+    if headerxml.findtext('stepDetails/DeviceClass', '') == 'STARSTIM':
+        print('Found Starstim, not sure how to handle this')
+
+    # Trigger data: 4 bytes in newer versions, 1 byte in older versions
+    trigger_type = '>i4' if headerxml.findtext('NEDFversion') else 'B'
+    datadt.append(('trig', trigger_type))
+    # 5 data samples per block
+    dt.append(('data', np.dtype(datadt), (5,)))
+
+    date = headerxml.findtext('StepDetails/StartDate_firstEEGTimestamp', '')
+    info['meas_date'] = datetime.datetime.utcfromtimestamp(int(date) / 1000)
+    return info, np.dtype(dt)
+
+
+def _read_nedf_eeg(filename: str):
+    """
+    Read header info and EEG data from an .nedf file
+
+    Parameters
+    ----------
+    filename : str
+        Path to the .nedf file.
+
+    Returns
+    -------
+    eeg : array, shape (n_samples, n_channels)
+        Unscaled EEG data
+    info : dict
+        Information from the file header
+    triggers : array, shape (n_annots, 2)
+        Start samples and values of each trigger
+    scale : float
+        Scaling factor for the EEG data
+    """
+    info, dt = parse_nedf_header(filename)
+
+    # to quote the original matlab implementation:
+    # "binary data will always start at byte 5120"
+    binstart = 10240
+    with open(filename, mode='rb') as f:
+        f.seek(binstart, os.SEEK_SET)
+        data = np.fromfile(f, dtype=dt)
+
+    # convert uint8-triplet -> float32
+    eeg = data['data']['eeg'] @ [1 << 16, 1 << 8, 1.]
+    eeg = eeg.reshape((-1, info['nchan']))
+    # convert sign if necessary
+    eeg[eeg > (1 << 23)] -= 1 << 24
+
+    triggers = data['data']['trig'].flatten()
+    triggerind = triggers.nonzero()[0]
+    triggers = np.stack((triggerind, triggers[triggerind])).T
+
+    # scale channels accordingly (here: to volts)
+    scale = 2.4 / (6.0 * 8388607)
+
+    return eeg, info, triggers, scale
+
+
+class RawNedf(BaseRaw):
+    def _read_segment_file(self, data, idx, fi, start, stop, cals, mult):
+        pass
+
+    def __init__(self, nedffile):
+        eeg, header, triggers, scale = _read_nedf_eeg(nedffile)
+        info = create_info(ch_names=header['ch_names'],
+                           sfreq=header['sfreq'],
+                           ch_types='eeg')
+
+        info['bads'] = []
+        info['meas_date'] = header['meas_date']
+
+        super().__init__(info, preload=(eeg * scale).T, filenames=nedffile)
+        self.set_montage('standard_1020')
```

Comment:
Does this work for every file?

Reply:
As far as I know the NeuroElectrics enobio/StarStim devices only have 10-10/10-20 caps, so until they sell caps with custom layouts it works.

Reply:
I undid this line about `set_montage`. If tomorrow they start selling different caps we have to go through a deprecation cycle to get rid of this `standard_1020` behavior or do something more complicated, and it's easy enough for users to do `read_raw_nedf(fname).set_montage('standard_1020')`.

One way I could think to keep this sort of line would be to have a check that if all of the EEG channels in `info` match the `standard_1020` names then we set it for them. But given discussions we've had about realistic versus pure-spherical electrode positions and the like, I don't mind people needing to add their own `set_montage` to be explicit about the montage they're using. It also allows potentially setting the head size, etc. -- sorts of things it might be good to have users think about / work through in some tutorials.